src/ConnectionPool.jl

```diff
@@ -483,59 +473,87 @@ function getconnection(::Type{Transaction{T}},
                        reuse_limit::Int=nolimit,
                        require_ssl_verification::Bool=true,
                        kw...)::Transaction{T} where T <: IO
-
-    while true
-
-        lock(poollock[Threads.threadid()])
-        try
-            # Close connections that have reached the reuse limit...
-            if reuse_limit != nolimit
-                for c in findoverused(T, host, port, reuse_limit)
-                    close(c)
-                end
+    pod = getpod(POOL, hashconn(T, host, port, pipeline_limit, require_ssl_verification, true))
```

Comment:
This does not look like a valid use of a `hash` function: you need to use the tuple of values, not their hash, as the lookup key.

Reply:
Can you explain more why this isn't a valid use of a `hash` function? The reason for using `hash` in my mind is to avoid allocating the tuple just to do the pod lookup, since we don't really need the tuple of values other than for looking up the right pod of connections.

Reply:
It's just not what a hash function does